client/app.go

```diff
@@ -1,62 +1,59 @@
 package client
 
 import (
-	"fmt"
-	"net/http"
-
-	apps "github.com/replicatedhq/replicated/gen/go/v1"
+	"github.com/replicatedhq/replicated/pkg/types"
 )
 
-// ListApps returns all apps and their channels.
-func (c *HTTPClient) ListApps() ([]apps.AppAndChannels, error) {
-	appsAndChannels := make([]apps.AppAndChannels, 0)
-	err := c.doJSON("GET", "/v1/apps", http.StatusOK, nil, &appsAndChannels)
+func (c *Client) ListApps() ([]types.AppAndChannels, error) {
+	platformApps, err := c.PlatformClient.ListApps()
 	if err != nil {
 		return nil, err
 	}
-	return appsAndChannels, nil
-}
 
-// GetApp resolves an app by either slug or ID.
-func (c *HTTPClient) GetApp(slugOrID string) (*apps.App, error) {
-	appsAndChannels, err := c.ListApps()
+	shipApps, err := c.ShipClient.ListApps()
 	if err != nil {
-		return nil, fmt.Errorf("GetApp: %v", err)
+		return nil, err
 	}
-	for _, ac := range appsAndChannels {
-		if ac.App.Slug == slugOrID || ac.App.Id == slugOrID {
-			return ac.App, nil
+
+	apps := make([]types.AppAndChannels, 0, 0)
+	for _, platformApp := range platformApps {
+		channels := make([]types.Channel, 0, 0)
+		for _, platformChannel := range platformApp.Channels {
+			channel := types.Channel{
+				ID:          platformChannel.Id,
+				Name:        platformChannel.Name,
+				Description: platformChannel.Description,
+			}
+
+			channels = append(channels, channel)
 		}
+
+		app := types.AppAndChannels{
+			App: &types.App{
+				ID:        platformApp.App.Id,
+				Name:      platformApp.App.Name,
+				Scheduler: platformApp.App.Scheduler,
+				Slug:      platformApp.App.Scheduler,
+			},
+			Channels: channels,
+		}
+
+		apps = append(apps, app)
 	}
-	return nil, ErrNotFound
+
+	apps = append(apps, shipApps...)
+
+	return apps, nil
 }
 
-// CreateApp creates a new app with the given name and returns it.
-func (c *HTTPClient) CreateApp(opts *AppOptions) (*apps.App, error) {
-	reqBody := &apps.Body{Name: opts.Name}
-	app := &apps.App{}
-	err := c.doJSON("POST", "/v1/app", http.StatusCreated, reqBody, app)
-	if err != nil {
-		return nil, err
-	}
-	return app, nil
+func (c *Client) GetApp(appID string) (interface{}, error) {
+	return nil, nil
 }
 
-// DeleteApp deletes an app by id.
-func (c *HTTPClient) DeleteApp(id string) error {
-	endpoint := fmt.Sprintf("%s/v1/app/%s", c.apiOrigin, id)
-	req, err := http.NewRequest("DELETE", endpoint, nil)
-	if err != nil {
-		return err
-	}
-	req.Header.Add("Authorization", c.apiKey)
-	resp, err := http.DefaultClient.Do(req)
-	if err != nil {
-		return fmt.Errorf("DeleteApp (%s %s): %v", req.Method, endpoint, err)
-	}
-	resp.Body.Close()
-	if resp.StatusCode != http.StatusNoContent {
-		return fmt.Errorf("DeleteApp (%s %s): status %d", req.Method, endpoint, resp.StatusCode)
-	}
+func (c *Client) CreateApp(appOptions interface{}) (interface{}, error) {
+	return nil, nil
```

Comment:
i assume you will fill these in before merging?

Reply:
Yes, but not a blocker here. The CreateApp function in the CLI still calls the platformapi method directly. My intent isn't to add all ship functionality in 1 PR, but to start adding it, as needed.